plugins/outputs/instrumental/instrumental.go

```diff
@@ -0,0 +1,154 @@
+	conn net.Conn
+}
+
+const (
+	DefaultHost = "collector.instrumentalapp.com"
+	AuthFormat  = "hello version go/telegraf/1.0\nauthenticate %s\n"
+)
+
+var sampleConfig = `
+  ## Project API Token (required)
+  api_token = "API Token" # required
+  ## Prefix the metrics with a given name
+  prefix = ""
+  ## Stats output template (Graphite formatting)
+  ## see https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md#graphite
+  template = "host.tags.measurement.field"
+  ## Timeout in seconds to connect
+  timeout = 2
+`
+
+func (i *Instrumental) Connect() error {
+	timeout := time.Duration(i.Timeout) * time.Second
+	connection, err := net.DialTimeout("tcp", i.Host+":8000", timeout)
+	if err != nil {
+		return err
+	}
+
+	err = i.authenticate(connection)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (i *Instrumental) Close() error {
+	i.conn.Close()
+	return nil
+}
+
+func (i *Instrumental) Write(metrics []telegraf.Metric) error {
+	s, err := serializers.NewGraphiteSerializer(i.Prefix, i.Template)
+	if err != nil {
+		return err
+	}
+
+	var points []string
+	var metricType string
+	var toSerialize telegraf.Metric
+	var newTags map[string]string
+
+	for _, metric := range metrics {
+		// Pull the metric_type out of the metric's tags. We don't want the type
+		// to show up with the other tags pulled from the system, as they go in the
+		// beginning of the line instead.
+		// e.g we want:
+		//
+		//  increment some_prefix.host.tag1.tag2.tag3.field value timestamp
+		//
+		// vs
+		//
+		//  increment some_prefix.host.tag1.tag2.tag3.counter.field value timestamp
+		//
+		newTags = metric.Tags()
+		metricType = newTags["metric_type"]
+		delete(newTags, "metric_type")
+
+		toSerialize, _ = telegraf.NewMetric(
+			metric.Name(),
+			newTags,
+			metric.Fields(),
+			metric.Time(),
+		)
+
+		stats, err := s.Serialize(toSerialize)
+		if err != nil {
+			log.Printf("Error serializing a metric to Instrumental: %s", err)
+		}
+
+		switch metricType {
+		case "counter":
+			fallthrough
+		case "histogram":
+			metricType = "increment"
+		case "set":
+			metricType = "gauge_absolute"
+		default:
+			metricType = "gauge"
+		}
+
+		for _, stat := range stats {
+			points = append(points, fmt.Sprintf("%s %s", metricType, stat))
+		}
+	}
+
+	allPoints := strings.Join(points, "\n") + "\n"
+
+	_, err = fmt.Fprintf(i.conn, allPoints)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (i *Instrumental) Description() string {
+	return "Configuration for sending metrics to an Instrumental project"
+}
+
+func (i *Instrumental) SampleConfig() string {
+	return sampleConfig
+}
+
+func (i *Instrumental) authenticate(conn net.Conn) error {
+	_, err := fmt.Fprintf(conn, AuthFormat, i.ApiToken)
+	if err != nil {
+		return err
+	}
+
+	i.conn = conn
+	return nil
+}
+
+func init() {
+	outputs.Add("instrumental", func() telegraf.Output {
+		return &Instrumental{
+			Host:     DefaultHost,
+			Template: graphite.DEFAULT_TEMPLATE,
```

Comment:
I wasn't sure what the best way to provide default values is here. Does this get over-written by the Config parsing or does any already-set value take precedence?